airflow/models.py

```diff
@@ -1117,6 +1149,9 @@ def __init__(
         self.pool = pool
         self.sla = sla
         self.execution_timeout = execution_timeout
+        self.on_failure_callback = on_failure_callback
+        self.on_success_callback = on_success_callback
+        self.on_retry_callback = on_retry_callback
```

Comment:
I think it muddies the API to expect a list or either a callable or a list of callable and to deal with using type inspection. If you need to call multiple things, write a function that calls multiple things, or wrap it in a lambda that calls something like this
```
def calls(funcs, *args, **args):
    return [f(*args, **args) for f in funcs]
```